modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/expcatalog/ExperimentSummaryRepository.java

```diff
@@ -0,0 +1,265 @@
+                }
+
+                else if (field.equals(DBConstants.Experiment.EXECUTION_ID)) {
+                    logger.debug("Filter Experiments by Execution ID");
+                    queryParameters.put(DBConstants.Experiment.EXECUTION_ID, filters.get(field));
+                    query += "ES.executionId LIKE :" + DBConstants.Experiment.EXECUTION_ID + " AND ";
+                }
+
+            }
+
+        }
+
+        if (filters.get(DBConstants.ExperimentStatus.STATE) != null) {
+            logger.debug("Filter Experiments by State");
+            String state = ExperimentState.valueOf(filters.get(DBConstants.ExperimentStatus.STATE)).toString();
+            queryParameters.put(DBConstants.ExperimentStatus.STATE, state);
+            query += "ES.state LIKE :" + DBConstants.ExperimentStatus.STATE + " AND ";
+        }
+
+        if (filters.get(DBConstants.ExperimentSummary.FROM_DATE) != null
+                && filters.get(DBConstants.ExperimentSummary.TO_DATE) != null) {
+
+            Timestamp fromDate = Timestamp.valueOf(filters.get(DBConstants.ExperimentSummary.FROM_DATE));
+            Timestamp toDate = Timestamp.valueOf(filters.get(DBConstants.ExperimentSummary.TO_DATE));
+
+            if (toDate.after(fromDate)) {
+                logger.debug("Filter Experiments by CreationTime");
+                queryParameters.put(DBConstants.ExperimentSummary.FROM_DATE, fromDate);
+                queryParameters.put(DBConstants.ExperimentSummary.TO_DATE, toDate);
+                query += "ES.creationTime BETWEEN :" + DBConstants.ExperimentSummary.FROM_DATE + " AND :" + DBConstants.ExperimentSummary.TO_DATE + " AND ";
+            }
+
+        }
+
+        if (accessibleExperimentIds != null && !accessibleExperimentIds.isEmpty()) {
+            logger.debug("Filter Experiments by Accessible Experiment IDs");
+            queryParameters.put(DBConstants.Experiment.ACCESSIBLE_EXPERIMENT_IDS, accessibleExperimentIds);
+            query += " ES.experimentId IN :" + DBConstants.Experiment.ACCESSIBLE_EXPERIMENT_IDS;
+        }
+
+        else {
+            logger.debug("Removing the last operator from the query");
+            query = query.substring(0, query.length() - 5);
+        }
+
+        List<ExperimentSummaryModel> experimentSummaryModelList = select(query, limit, offset, queryParameters);
+        return experimentSummaryModelList;
+    }
+
+    public ExperimentStatistics getExperimentStatistics(Map<String,String> filters) throws RegistryException {
+
+        try {
+
+            ExperimentStatistics experimentStatistics = new ExperimentStatistics();
+            String gatewayId = null;
+            String userName = null;
+            String applicationName = null;
+            String resourceHostName = null;
+            Timestamp fromDate = null;
+            Timestamp toDate = null;
+
+            for (String field : filters.keySet()) {
+
+                if (field.equals(DBConstants.Experiment.GATEWAY_ID)) {
```

Comment:
I think GATEWAY_ID is required so it should be handled differently and an exception should be thrown if it isn't in the list of filters.

Reply:
Done.